### PR TITLE
[MAINTENANCE] Handle Pandas warnings in Data Assistant plots

### PR DESCRIPTION
**Pull request by NathanFarmer**
Changes proposed in this pull request:
- GREAT-1175
- Handle Pandas warnings in Data Assistant plots
- Get rid of text representation of PlotResult

### Definition of Done
- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [X] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/rule_based_profiler/data_assistant_result/plot_result.py [hunk 1]
```diff
@@ -24,3 +24,6 @@ class PlotResult:
     """
 
     charts: List[alt.Chart]
+
+    def __repr__(self):
+        return ""
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/data_assistant_result/plot_result.py:29]: Could we annotate with `-> str`. Also, is this the recommend way to remove a repr?

**Review by cdkini — APPROVED** [review]

File: great_expectations/rule_based_profiler/data_assistant_result/data_assistant_result.py [hunk 2]
```diff
@@ -2492,16 +2474,7 @@ def _get_interactive_range_chart(
...
-            input_dropdown_initial_state: pd.DataFrame = df.groupby(
-                [batch_plot_component.name], as_index=False
-            ).max()
-            input_dropdown_initial_state[
-                batch_plot_component.batch_identifiers + [domain_plot_component.name]
-            ] = " "
-            df = pd.concat([input_dropdown_initial_state, df], axis=0)
-
-        columns: List[str] = pd.unique(df[domain_plot_component.name]).tolist()
+        columns: List[str] = [""] + pd.unique(df[domain_plot_component.name]).tolist()
```
Comment (Shinnnyshinshin) [inline great_expectations/rule_based_profiler/data_assistant_result/data_assistant_result.py:2477]: ❤️

**Review by Shinnnyshinshin — APPROVED** [review]
Love it. Thank you @NathanFarmer

File: great_expectations/rule_based_profiler/data_assistant_result/plot_result.py (hunk 1 above)
Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/data_assistant_result/plot_result.py:29]: I updated the annotation! ✅

Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/data_assistant_result/plot_result.py:29]: As far as I can find this is the preferred method, but please let me know if you see something else!

**Pull request merged**